dss/storage/checkout.py

```diff
@@ -23,6 +26,90 @@ class ValidationEnum(Enum):
     PASSED = auto()
 
 
+class BundleNotFoundError(Exception):
+    """Raised when we attempt to check out a non-existent bundle."""
+    pass
+
+
+class CheckoutTokenKeys:
+    """
+    When we are executing a request that involves a checkout, the client will periodically check back in to see if the
+    checkout is complete.  To avoid duplicating checkout requests, the client will check back using a token.  These are
```

Comment:
It seems like the deduplication is only done by checking that the token exists, the only key-value pair touched is `attempts`, and that value isn't leveraged. Seems like a simple flag would perform the same function(?) Is there a dev UX element here?

Reply:
It's for future use.  If time has elapsed very little, but attempts is high, that means we have a client that's not respecting the Retry-After header.  We could also implement exponential backoff or other creative solutions for managing retries.